### PR TITLE
chore: Add alternate defaults for mysql & postgresql blueprint

### PR DESCRIPTION
**Pull request by qz267**
More details in http://docs/document/d/1LMsDRy0zMKQrPUDJ4chlPtqul9bwXs90QH6N8xqZ78I?tab=t.0#heading=h.qx5xjbm73vyx

### CODE REVIEW

**Comment by imrannayer** [comment]
@qz267 is there any reason for changing defaults instead of sending values and override it in module call?

**Comment by qz267** [comment]
> @qz267 is there any reason for changing defaults instead of sending values and override it in module call?

Sorry for the confusion, I was trying to surface some of my WIP changes, to ask for early feedbacks, PR title updated.

b/362377949 is the original request, I don't have strong preference about update in variables or override them in the module call, any suggestions are welcomed, thanks!

**Comment by imrannayer** [comment]
@qz267 I think it will be better to take alternate route instead of changing defaults in the module as this change will require all the existing user of the module to adjust code for new defaults.

@bharathkkb can u plz review?

**Comment by qz267** [comment]
I got @q2w suggested using `alt_defaults` instead https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/cli/bpmetadata/proto/bpmetadata_ui.proto#L149, WDYT?

File: modules/mysql/metadata.display.yaml [hunk 1]
```diff
@@ -43,6 +43,9 @@ spec:
         backup_configuration:
           name: backup_configuration
           title: Backup Configuration
+          alt_defaults:
+            - type: ALTERNATE_TYPE_UNSPECIFIED
```
Comment (bharathkkb) [inline modules/mysql/metadata.display.yaml:47]: I think we want these to be `ALTERNATE_TYPE_DC` but will let @q2w confirm

Comment (q2w) [reply modules/mysql/metadata.display.yaml:47]: @qz267 Let's use `ALTERNATE_TYPE_DC ` which we have introduced for ADC!

Comment (qz267) [reply modules/mysql/metadata.display.yaml]: DONE

File: modules/postgresql/metadata.display.yaml [hunk 2]
```diff
@@ -40,9 +40,15 @@ spec:
...
           title: Availability Type
+          alt_defaults:
+            - type: ALTERNATE_TYPE_DC
+              value: REGIONAL
         backup_configuration:
           name: backup_configuration
           title: Backup Configuration
+          alt_defaults:
+            - type: ALTERNATE_TYPE_DC
+              value: true
```
[the remark this answers is not in the document]
Comment (qz267) [reply modules/postgresql/metadata.display.yaml]: Updated in https://github.com/terraform-google-modules/terraform-google-sql-db/pull/646/commits/dba5ec152cf7f3c051eac7b29dce889257d8102b, let me know how it looks for you.

File: modules/postgresql/metadata.display.yaml
Comment (q2w) [inline modules/postgresql/metadata.display.yaml:1]: `data_cache_enabled ` should also have alt_defaults set to `true`

File: modules/postgresql/metadata.display.yaml [hunk 3]
```diff
@@ -58,6 +68,11 @@ spec:
         database_flags:
           name: database_flags
           title: Database Flags
+          alt_defaults:
+            - type: ALTERNATE_TYPE_DC
+              value:
+                - name: "cloudsql.iam_authentication",
```
Comment (q2w) [inline modules/postgresql/metadata.display.yaml:74]:
```suggestion
                - name: "cloudsql.iam_authentication"
```

File: modules/postgresql/metadata.display.yaml [hunk 4]
```diff
@@ -40,9 +40,19 @@ spec:
...
           title: Availability Type
+          alt_defaults:
+            - type: ALTERNATE_TYPE_DC
+              value: REGIONAL
         backup_configuration:
           name: backup_configuration
           title: Backup Configuration
+          alt_defaults:
+            - type: ALTERNATE_TYPE_DC
+              value:
```
Comment (q2w) [inline modules/postgresql/metadata.display.yaml:51]: The value should be yaml equivalent of `{
  "enabled": true,
  "point_in_time_recovery_enabled": true
}`. I guest it should be like

```
value:
  enabled: true
  point_in_time_recovery_enabled: true
```

File: modules/mysql/metadata.display.yaml [hunk 5]
```diff
@@ -52,9 +60,17 @@ spec:
...
+          alt_defaults:
+            - type: ALTERNATE_TYPE_DC
+              value: true
         database_flags:
           name: database_flags
           title: Database Flags
+          alt_defaults:
+            - type: ALTERNATE_TYPE_DC
+              value:
+                - name: "cloudsql_iam_authentication",
```
Comment (q2w) [inline modules/mysql/metadata.display.yaml:72]:
```suggestion
                - name: "cloudsql_iam_authentication"
```

**Review by q2w — APPROVED** [review]

**Pull request merged**